.github/workflows/build_and_test.yml

```diff
@@ -268,6 +269,11 @@ jobs:
         export TERM=vt100
         # Hive "other tests" test needs larger metaspace size based on experiment.
         if [[ "$MODULES_TO_TEST" == "hive" ]] && [[ "$EXCLUDED_TAGS" == "org.apache.spark.tags.SlowHiveTest" ]]; then export METASPACE_SIZE=2g; fi
+        // SPARK-46283: should delete the following env replacement after SPARK 3.x EOL
+        if [[ "$MODULES_TO_TEST" == *"streaming-kinesis-asl"* ]] && [[ "BRANCH" != "master-1" ]]; then 
+          MODULES_TO_TEST=${MODULES_TO_TEST//streaming-kinesis-asl, /}
+        fi
+        echo $MODULES_TO_TEST
```

Comment:
will revet [b7acf54](https://github.com/apache/spark/pull/44204/commits/b7acf54fafb1204f1a7f1655a15db2ba86feaecb) after test

Reply:
![image](https://github.com/apache/spark/assets/1475305/d36172ce-0b5c-4f8a-947b-d34d1780ccc6)

From the printed `$MODULES_TO_TEST`, `streaming-kinesis-asl,` is removed when the condition is true.

let me revert b7acf54